src/utils.py

```diff
@@ -47,10 +50,46 @@ def safe_get_file(filepath: str) -> Optional[List[str]]:
     return content
 
 
-def generate_password():
+def generate_password() -> str:
     """Creates randomized string for use as app passwords.
 
     Returns:
         String of 32 randomized letter+digit characters
     """
     return "".join([secrets.choice(string.ascii_letters + string.digits) for _ in range(32)])
+
+
+def map_env(env: list[str]) -> dict[str, str]:
+    """Builds environment map for arbitrary env-var strings.
+
+    Returns:
+        Dict of env-var and value
+    """
+    map_env = {}
+    for var in env:
+        key = "".join(var.split("=", maxsplit=1)[0])
+        value = "".join(var.split("=", maxsplit=1)[1:])
+
+        map_env[key] = value
+
+    return map_env
+
+
+def get_env() -> dict[str, str]:
+    """Builds map of current basic environment for all processes.
+
+    Returns:
+        Dict of env-var and value
+    """
+    raw_env = safe_get_file("/etc/environment") or []
+    return map_env(env=raw_env)
+
+
+def update_env(env: dict[str, str]) -> None:
+    """Updates /etc/environment file."""
+    logger.info(f"{env=}")
+    logger.info(f"{get_env()=}")
+    updated_env = {**get_env(), **env}
+    logger.info(f"{updated_env=}")
+    content = "\n".join([f"{key}={value}" for key, value in updated_env.items()])
```

Comment:
If `updated_env` has the same content of `env` do we really need to rewrite the file ?

Reply:
I don't see why not, it doesn't harm anything no?